postmaster/models.py

```
@@ -366,6 +371,17 @@ def set_password(self, new_password):
 
         self.password = bcrypt.generate_password_hash(new_password)
 
+    def generate_otp_secret(self, **kwargs):
+        # generate a random secret
+        self.otp_secret = base64.b32encode(urandom(10)).decode('utf-8')
+
+    def get_totp_uri(self):
```

This should probably throw an exception if self.otp_secret is not set